.ci/atime/tests.R

```diff
@@ -208,5 +208,20 @@ test.list <- atime::atime_test_list(
     Slow = "fd24a3105953f7785ea7414678ed8e04524e6955", # Parent of the merge commit (https://github.com/Rdatatable/data.table/commit/ed72e398df76a0fcfd134a4ad92356690e4210ea) of the PR (https://github.com/Rdatatable/data.table/pull/5054) that fixes the issue
     Fast = "ed72e398df76a0fcfd134a4ad92356690e4210ea"), # Merge commit of the PR (https://github.com/Rdatatable/data.table/pull/5054) that fixes the issue
 
+# Improvement discussed in and brought by https://github.com/Rdatatable/data.table/pull/4386
+"forderv improved in #4386" = atime::atime_test(
+  N = 10^seq(3, 8), # 1e9 exceeds the runner's memory (process gets killed)
+  setup = {
+    options(datatable.forder.auto.index = TRUE, datatable.forder.reuse.sorting = TRUE)
```

Comment:
I wonder if setting these options will affect other tests? 
I checked where this is defined / documented and I found for `auto.index` :

```
grep  -nH --null "auto[.]index" ../*/*
../R/data.table.R:3309:    if (!getOption("datatable.auto.index")) return(NULL)
../R/onLoad.R:87:       "datatable.auto.index"="TRUE",          # DT[col=="val"] to auto add index so 2nd time faster
../R/setkey.R:144:  if (isTRUE(getOption("datatable.forder.auto.index"))) return(invisible())
grep: ../inst/include: Is a directory
grep: ../inst/po: Is a directory
grep: ../inst/tests: Is a directory
../man/datatable-optimize.Rd:102:\code{options(datatable.auto.index = FALSE)}. To switch off using existing
Binary file ../src/data_table.dll matches
../src/forder.c:1628:// isTRUE(getOption("datatable.auto.index"))
../src/forder.c:1630:  // for now temporarily 'forder.auto.index' not 'auto.index' to disabled it by default
../src/forder.c:1633:  SEXP opt = GetOption(install("datatable.forder.auto.index"), R_NilValue);
../src/forder.c:1637:    error("'datatable.forder.auto.index' option must be TRUE or FALSE"); // # nocov
../src/forder.c:1771:        GetAutoIndex()) { // disabled by default, use datatable.forder.auto.index=T to enable, do not export/document, use for debugging only
Binary file ../src/forder.o matches
grep: ../vignettes/css: Is a directory
../vignettes/datatable-benchmarking.Rmd:67:options(datatable.auto.index=TRUE)
../vignettes/datatable-benchmarking.Rmd:72:- `auto.index=FALSE` disables building index automatically when doing subset on non-indexed data, but if indices were created before this option was set, or explicitly by calling `setindex` they still will be used for optimization.
../vignettes/datatable-secondary-indices-and-auto-indexing.Rmd:318:* Auto indexing can be disabled by setting the global argument `options(datatable.auto.index = FALSE)`.
grep: ../vignettes/plots: Is a directory
```
above I see two options, `datatable.forder.auto.index` and `datatable.auto.index`
* `datatable.auto.index` is documented in Rd and two vignettes
* forder.c comments say "datatable.forder.auto.index=T to enable, do not export/document, use for debugging only"

which option does your test require setting? can you please add a comment to clarify why?


and I found for `reuse.sorting`:
```
grep  -nH --null "reuse[.]sorting" ../*/*
../R/setkey.R:151:forderv = function(x, by=seq_along(x), retGrp=FALSE, retStats=retGrp, sort=TRUE, order=1L, na.last=FALSE, reuseSorting=getOption("datatable.reuse.sorting", NA)) {
```

in your test you have `datatable.forder.reuse.sorting` but in the code there is no forder in the option name, `datatable.reuse.sorting` ... so I guess this option is not required to get your test result... where did you get that option name?

Reply:
Yes I don't need `datatable.reuse.sorting`, so do you want me to add a comment as to why `datatable.forder.auto.index` is required?

Reply:
> I wonder if setting these options will affect other tests?

I hope not! (we'll have to see from the results we get here)

Reply:
Doesn't look like its affecting the other tests